python-sdk/nuscenes/map_expansion/map_api.py

```diff
@@ -436,6 +453,105 @@ def get_bounds(self, layer_name: str, token: str) -> Tuple[float, float, float,
+        :return: Dictionary of layer_name - tokens pairs.
+        """
+
+        patch = (x - radius, y - radius, x + radius, y + radius)
+        return self.explorer.get_records_in_patch(patch, layer_names, mode)
+
+    def discretize_lanes(self, tokens: List[str],
+                         resolution_meters: float) -> Dict[str, List[Tuple[float, float, float]]]:
+        """
+        Discretizes a list of lane/lane connector tokens.
+        :param tokens: List of lane and/or lane connector record tokens. Can be retrieved with
+            get_records_in_radius or get_records_in_patch.
+        :param resolution_meters: How finely to discretize the splines.
+        :return: Mapping from lane/lane connector token to sequence of poses along the lane.
+        """
+
+        return {ID: discretize_lane(self.arcline_path_3.get(ID, []), resolution_meters) for ID in tokens}
+
+    def _get_connected_lanes(self, lane_token: str, incoming_outgoing: str) -> List[str]:
+        """
+        Helper for getting the lanes connected to a given lane
+        :param lane_token: Token for the lane.
+        :param incoming_outgoing: Whether to get incoming or outgoing lanes
+        :return: List of lane tokens this lane is connected to.
+        """
+
+        if lane_token not in self.connectivity:
+            raise ValueError(f"{lane_token} is not a valid lane.")
+
+        return self.connectivity[lane_token][incoming_outgoing]
+
+    def get_outgoing_lane_ids(self, lane_token: str) -> List[str]:
+        """
+        Get the out-going lanes
+        :param lane_token: Token for the lane
+        :return: List of lane tokens that start at the end of this lane.
+        """
+
+        return self._get_connected_lanes(lane_token, 'outgoing')
+
+    def get_incoming_lane_ids(self, lane_token: str) -> List[str]:
+        """
+        Get the incoming lanes
+        :param lane_token: Token for the lane
+        :return: List of lane tokens that end at the start of this lane.
+        """
+
+        return self._get_connected_lanes(lane_token, 'incoming')
+
+    def get_lane(self, lane_token: str) -> List[ArcLinePath]:
+        """
+        Get the arc line path representation for a lane.
+        :param lane_token: Token for the lane
+        :return: Arc line path representation of the lane
+        """
+
+        lane = self.arcline_path_3.get(lane_token)
+        if not lane:
+            raise ValueError(f'Lane token {lane_token} is not a valid lane.')
+
+        return lane
+
+    def get_closest_lane(self, x: float, y: float) -> str:
+        """
+        Get closest lane id. The distance from a point (x, y) to a lane is the minimum l2 distance
+        from (x, y) to a point on the lane
+        :param x: X coordinate in global coordinate frame
+        :param y: Y Coordinate in global coordinate frame
+        :return: Lane id of closest lane
+        """
+
+        lanes = self.get_records_in_radius(x, y, 5, ['lane', 'lane_connector'])
```

Comment:
Make the radius a parameter of `get_closest_lane()`

Reply:
Done.